File: content/problems/1346-check-if-n-and-its-double-exist/solutions/seen-double-or-half/Solution.java
class Solution {
    public boolean checkIfExist(int[] arr){
        Set<Integer> seen=new HashSet<>();
        for(int value:arr){
            if(seen.contains(value*2)||value%2==0&&seen.contains(value/2))return true;
            seen.add(value);
        }
        return false;
    }
}

File: content/problems/1346-check-if-n-and-its-double-exist/solutions/seen-double-or-half/Solution.kt
class Solution {
    fun checkIfExist(arr:IntArray):Boolean{
        val seen=mutableSetOf<Int>();
        for(value in arr){
            if(value*2 in seen||value%2==0&&value/2 in seen)return true;
            seen.add(value)
        };
        return false
    }
}
